Autopopulate 1RM

xenogerm opened the pull request:
**Current State:**
When using the 1RM (Rep Max) calculator, known values are typically 5 x 10 x 135.
<img width="628" height="344" alt="image" src="https://github.com/user-attachments/assets/8f9f3f1b-5ec7-4d81-a466-d000a13aba4b" />


**Changes made with the PR**
If an exercise has an 1RM defined, it uses the data available to prepopulate with the 1RM set. We only use the Reps and Weight, assuming this was the "best" the user could do and thus does not need RPE value. The user can then click "Use it" or define their own values, but this saves 1 step.
<img width="662" height="134" alt="image" src="https://github.com/user-attachments/assets/07273c5f-f197-4090-be02-ddc43fa8d6ac" />
<img width="622" height="336" alt="image" src="https://github.com/user-attachments/assets/b3990ab3-1b21-430b-81da-6b1d57f6b50a" />

xenogerm committed:
Autopopulate 1RM

This autopopulates the repmaxcalculator with real values.

xenogerm committed:
Update repMaxCalculator.tsx

xenogerm committed:
Add correct types

Change  the types to make more sense

astashov commented on src/stores/oneRMStore.ts:
```diff
@@ -0,0 +1,45 @@
+};
+
+const data = new Map<string, OneRMTriplet>();
+const subs = new Map<string, Set<(v: OneRMTriplet | undefined) => void>>();
```
I frankly don't like this, sorry. Keeping state in global context is error-prone, and works poorly in tests.

astashov commented:
Ugh, that's a tricky problem.

Historically, I was trying to avoid any implicit passing data into the components. Like not even using React context. That helps a lot if you need to memoize the component for performance - if the root component has all the props for child components, it will be properly memoized and won't cause issues for child components. There's no hidden dependencies at all.

The problem here is since it goes into the keyboard, and needs pretty much the whole history to calculate the e1RMs.
So, we have 2 options:

* Pass `props.history` everywhere we need the RepMaxCalculator
* Calculate e1RMs in the `<App />`, and pass that value to the leafs. But that means we'll calculate it every time, even if we don't need it yet - it's a performance hit.

Another thing that the Custom Keyboard is really a React portal, so maaaybe there's a way to pass a prop without passing it through all the intermediate components. Need to look into it.

For now, let's try to explicitly pass `props.history` to RepMaxCalculator, and see how bad it gets. Maybe it's not THAT bad, and only need to be passed through few screens...

xenogerm committed:
Refactor OneRMSet

Removes the oneRMStore as it does not align with the product and would introduce test issues and unneeded complexity with saving global state.
